libs/sdk-ui-dashboard/src/model/commandHandlers/alerts/removeAlertsHandler.ts

```diff
@@ -0,0 +1,32 @@
+// (C) 2021 GoodData Corporation
+import { SagaIterator } from "redux-saga";
+import { call, select, put } from "redux-saga/effects";
+import { ObjRef, objRefToString } from "@gooddata/sdk-model";
+import { DashboardContext } from "../../types/commonTypes";
+import { RemoveAlerts } from "../../commands/alerts";
+import { alertsRemoved, DashboardAlertsRemoved } from "../../events/alerts";
+import { alertsActions } from "../../state/alerts";
+import { selectAlertsMap } from "../../state/alerts/alertsSelectors";
+import { validateExistingAlerts } from "./validation/alertsValidation";
+import { batchActions } from "redux-batched-actions";
+
+function removeAlerts(ctx: DashboardContext, alertRefs: ObjRef[]): Promise<void> {
+    const { backend, workspace } = ctx;
+    return backend.workspace(workspace).dashboards().deleteWidgetAlerts(alertRefs);
+}
+
+export function* removeAlertsHandler(
+    ctx: DashboardContext,
+    cmd: RemoveAlerts,
+): SagaIterator<DashboardAlertsRemoved> {
+    const alertsMap: ReturnType<typeof selectAlertsMap> = yield select(selectAlertsMap);
+    const alerts = validateExistingAlerts(alertsMap, cmd, ctx);
+
+    const refs = alerts.map((alert) => alert.ref);
+
+    yield call(removeAlerts, ctx, refs);
+
+    yield put(batchActions(alerts.map((alert) => alertsActions.removeAlert(objRefToString(alert.ref)))));
```

Comment:
I suggest to expose the removeMany entity adapter reducer and use it instead of this map. you can add it in state/alerts/index.ts. `removeAlerts: alertsAdapter.removeMany`.

there will still be need for the mapping because you need to get list of stringified refs, but no need for multiple reducer actions & their batching.

Reply:
Done